tests/Doctrine/Tests/Models/CMS/CmsAddress.php

```diff
@@ -91,7 +91,7 @@ class CmsAddress
 
     /**
      * @OneToOne(targetEntity="CmsUser", inversedBy="address")
-     * @JoinColumn(referencedColumnName="id")
+     * @JoinColumn(referencedColumnName="id", unique=false)
```

Comment:
Just a follow up since I see that there hasn't been any further discussion on this--  non-unique one to one is NOT quite the same as many to one at the application level because in the latter case the many side is represented by a subclass of Collection while in a one to zero or one there is no collection on either side.  It's important to remember that we are dealing with mapping between objects and database relations and because of the infamous impedance mismatch sometimes we have have to make leeway for things that might seem "wrong".   A better solution would be to have a more sophisticated UnitOfWork that could handle executing orphanRemoval deletes prior to inserts for associations within the bounds of a flush.  I'm happy to tackle this at some future time but I don't have time right now, so in the meantime I thought that maybe this patch would be acceptable.  In my project I only need this in the 2.3 code line anyway.....